prometheus/vec.go

```diff
@@ -25,10 +25,27 @@ import (
 // provided in this package.
 type MetricVec struct {
 	mtx      sync.RWMutex // Protects the children.
-	children map[uint64]Metric
+	children map[uint64][]metricLabelValue
 	desc     *Desc
 
 	newMetric func(labelValues ...string) Metric
+	hashAdd   func(h uint64, s string) uint64 // replace hash function for testing collision handling
```

Comment:
Having _everything_ collide exercises the code paths more completely. We'd have to find 3 or more collisions to test 2+N search case.

I'll use the collision provided in another test to make sure we are falling back correctly.